src/_blog/cryptotls.md

```diff
@@ -0,0 +1,60 @@
+---
+tags:
+- quic
+- go
+- standard library
+- crypto/tls
+title: A QUIC API for crypto/tls
+description:
+date: 2023-08-15
+permalink: "/2023-08-15-quic-crypto-tls/"
+translationKey: ''
+header_image: /quic-crypto-tls.png
+author: Marten Seemann
+---
+
+# A QUIC API for crypto/tls
+
+You might have heard us talk about this before: QUIC has become the most important transport in libp2p. For example, in the IPFS network, QUIC connections account for a whopping 80-90% of the connections. go-libp2p uses [quic-go](https://github.com/quic-go/quic-go), a QUIC implementation written in pure Go. quic-go doesn’t only power QUIC support in go-libp2p, but also in [Caddy](https://caddyserver.com/), a full-featured webserver (for its HTTP/3 support), [Adguard](https://github.com/AdguardTeam/AdGuardHome) (for providing DNS over QUIC), and in the synchronization tool [syncthing](https://github.com/syncthing/syncthing/), and [many other projects](https://github.com/quic-go/quic-go#projects-using-quic-go).
+
+## QUIC and TLS 1.3
+
+QUIC uses TLS 1.3 to secure the connection. There’s no such thing as an unencrypted QUIC connection! However, due to running on top of (the unreliable, unordered) UDP, QUIC's  interactions with the TLS stack is quite different from how a TLS connection on top of TCP would look like. The details are described in [RFC 9001](https://www.rfc-editor.org/rfc/rfc9001.html). When QUIC was standardized, it became a necessity for all TLS stacks across languages to expose new APIs. For the longest time, the Go standard library TLS package (crypto/tls) lacked an API for this purpose. We had no choice but to fork crypto/tls to add the required APIs ourselves.
```

Comment:
```suggestion
QUIC uses TLS 1.3 to secure the connection. There’s no such thing as an unencrypted QUIC connection! However, due to running on top of UDP (unreliable and unordered), QUIC's interactions with the TLS stack differs from how a TLS connection on top of TCP would look like. The details are described in [RFC 9001](https://www.rfc-editor.org/rfc/rfc9001.html). When QUIC was standardized, it became a necessity for all TLS stacks across languages to expose new APIs. For the longest time, the Go standard library TLS package (crypto/tls) lacked an API for this purpose. quic-go had no choice but to fork crypto/tls to add the required APIs ourselves.
```

> we had no choice

to "quic-go had no choice"
On a related note: the acknowledgment that you maintain quic-go should be added (perhaps in the intro)

> fork crypto/tls

link?